### ISSUE TITLE
Unable to execute

### ISSUE THREAD

**Issue by debanjansen48** [Valdes-Tresanco-MS/AMDock#12]
After installation, during execution received following error

(base) debanjan@debanjan:~/AM-Dock/AMDock$ bash AMDock.sh 
Traceback (most recent call last):
  File "/home/debanjan/AM-Dock/AMDock/AMDock/Docking_Program.py", line 3, in <module>
    from checker import Checker
  File "/home/debanjan/AM-Dock/AMDock/AMDock/checker.py", line 1, in <module>
    from warning import (prot_with_metal_warning, metal_no_ADzn_warning1,
  File "/home/debanjan/AM-Dock/AMDock/AMDock/warning.py", line 1, in <module>
    from PyQt4 import QtGui
ModuleNotFoundError: No module named 'PyQt4'

**Comment by Valdes-Tresanco-MS** [comment]
Hi @debanjansen48 
You must install the following packages before running it:
    
     sudo apt install pymol python-qt4 python-numpy openbabel

**Comment by debanjansen48** [comment]
I have that before sir.

(base) debanjan@debanjan:~$ sudo apt install  python-qt4
[sudo] password for debanjan:         
Reading package lists... Done
Building dependency tree       
Reading state information... Done
python-qt4 is already the newest version (4.12.1+dfsg-2).
python-qt4 set to manually installed.
The following packages were automatically installed and are no longer required:
  avogadro-data cuda-license-10-2 gdal-bin gdal-data grass-doc lib32gcc1
  lib32stdc++6 libarmadillo8 libarpack2 libavogadro1 libcharls1
  libclang-6.0-dev libclang-common-6.0-dev libclang-dev libdap25
  libdapclient6v5 libepsilon1 libfcgi0ldbl libfreexl1 libfyba0 libgdal20
  libgdcm2.8 libgeos-3.6.2 libgeos-c1v5 libgeotiff2 libgl2ps1.4 libgtkglext1
  libhdf4-0-alt libhdf5-openmpi-100 libjs-jquery-ui libjs-leaflet libkmlbase1
  libkmldom1 libkmlengine1 liblas-c3 liblas3 libllvm9 libllvm9:i386
  libminizip1 libnetcdf-c++4 libobjc-7-dev libobjc4 libodbc1 libogdi3.2
  libopenbabel4v5 libopencv-calib3d3.2 libopencv-contrib3.2 libopencv-core3.2
  libopencv-features2d3.2 libopencv-flann3.2 libopencv-highgui3.2
  libopencv-imgcodecs3.2 libopencv-imgproc3.2 libopencv-ml3.2
  libopencv-objdetect3.2 libopencv-photo3.2 libopencv-shape3.2
  libopencv-stitching3.2 libopencv-superres3.2 libopencv-video3.2
  libopencv-videoio3.2 libopencv-videostab3.2 libopencv-viz3.2 libpangox-1.0-0
  libproj12 libqca2 libqca2-plugins libqgis-analysis2.18.17 libqgis-app2.18.17
  libqgis-core2.18.17 libqgis-customwidgets libqgis-gui2.18.17
  libqgis-networkanalysis2.18.17 libqgis-server2.18.17 libqgisgrass7-2.18.17
  libqgispython2.18.17 libqhull7 libqjson0 libqscintilla2-qt4-13
  libqscintilla2-qt4-l10n libqt5positioning5 libqt5qml5 libqt5quick5
  libqt5sensors5 libqt5webchannel5 libqt5webkit5 libqwt6abi1 libsocket++1
  libspatialindex4v5 libspatialite7 libsuperlu5 libtbb2 liburiparser1
  libvtk6.3 libxerces-c3.2 linux-headers-5.3.0-51
  linux-headers-5.3.0-51-generic linux-headers-5.3.0-53
  linux-headers-5.3.0-53-generic linux-headers-5.3.0-59
  linux-headers-5.3.0-59-generic linux-headers-5.3.0-61
  linux-headers-5.3.0-61-generic linux-headers-5.4.0-52-generic
  linux-headers-5.4.0-53-generic linux-headers-5.4.0-54-generic
  linux-headers-5.4.0-56-generic linux-headers-5.4.0-58-generic
  linux-headers-5.4.0-59-generic linux-headers-5.4.0-60-generic
  linux-headers-5.4.0-62-generic linux-headers-5.4.0-64-generic
  linux-headers-5.4.0-65-generic linux-headers-5.4.0-66-generic
  linux-headers-5.4.0-67-generic linux-headers-5.4.0-70-generic
  linux-headers-5.4.0-71-generic linux-hwe-5.4-headers-5.4.0-52
  linux-hwe-5.4-headers-5.4.0-53 linux-hwe-5.4-headers-5.4.0-54
  linux-hwe-5.4-headers-5.4.0-56 linux-hwe-5.4-headers-5.4.0-58
  linux-hwe-5.4-headers-5.4.0-59 linux-hwe-5.4-headers-5.4.0-60
  linux-hwe-5.4-headers-5.4.0-62 linux-hwe-5.4-headers-5.4.0-64
  linux-hwe-5.4-headers-5.4.0-65 linux-hwe-5.4-headers-5.4.0-66
  linux-hwe-5.4-headers-5.4.0-67 linux-hwe-5.4-headers-5.4.0-70
  linux-hwe-5.4-headers-5.4.0-71 linux-image-5.3.0-51-generic
  linux-image-5.3.0-53-generic linux-image-5.3.0-59-generic
  linux-image-5.3.0-61-generic linux-image-5.3.0-62-generic
  linux-image-5.4.0-52-generic linux-image-5.4.0-53-generic
  linux-image-5.4.0-54-generic linux-image-5.4.0-56-generic
  linux-image-5.4.0-58-generic linux-image-5.4.0-59-generic
  linux-image-5.4.0-60-generic linux-image-5.4.0-62-generic
  linux-image-5.4.0-64-generic linux-image-5.4.0-65-generic
  linux-image-5.4.0-66-generic linux-image-5.4.0-67-generic
  linux-image-5.4.0-70-generic linux-image-5.4.0-71-generic
  linux-modules-5.3.0-51-generic linux-modules-5.3.0-53-generic
  linux-modules-5.3.0-59-generic linux-modules-5.3.0-61-generic
  linux-modules-5.3.0-62-generic linux-modules-5.4.0-52-generic
  linux-modules-5.4.0-53-generic linux-modules-5.4.0-54-generic
  linux-modules-5.4.0-56-generic linux-modules-5.4.0-58-generic
  linux-modules-5.4.0-59-generic linux-modules-5.4.0-60-generic
  linux-modules-5.4.0-62-generic linux-modules-5.4.0-64-generic
  linux-modules-5.4.0-65-generic linux-modules-5.4.0-66-generic
  linux-modules-5.4.0-67-generic linux-modules-5.4.0-70-generic
  linux-modules-5.4.0-71-generic linux-modules-extra-5.3.0-51-generic
  linux-modules-extra-5.3.0-53-generic linux-modules-extra-5.3.0-59-generic
  linux-modules-extra-5.3.0-61-generic linux-modules-extra-5.3.0-62-generic
  linux-modules-extra-5.4.0-52-generic linux-modules-extra-5.4.0-53-generic
  linux-modules-extra-5.4.0-54-generic linux-modules-extra-5.4.0-56-generic
  linux-modules-extra-5.4.0-58-generic linux-modules-extra-5.4.0-59-generic
  linux-modules-extra-5.4.0-60-generic linux-modules-extra-5.4.0-62-generic
  linux-modules-extra-5.4.0-64-generic linux-modules-extra-5.4.0-65-generic
  linux-modules-extra-5.4.0-66-generic linux-modules-extra-5.4.0-67-generic
  linux-modules-extra-5.4.0-70-generic linux-modules-extra-5.4.0-71-generic
  proj-data python-appdirs python-backports.functools-lru-cache python-certifi
  python-colorama python-cycler python-dateutil python-decorator
  python-easydev python-jinja2 python-markupsafe python-matplotlib-data
  python-owslib python-pexpect python-psycopg2 python-ptyprocess
  python-pyparsing python-pyproj python-pyspatialite python-qscintilla2
  python-qt4-sql python-requests python-subprocess32 python-tz python-yaml
  qgis-common qgis-plugin-grass-common qgis-providers qgis-providers-common
  qml-module-qtgraphicaleffects qml-module-qtquick-controls
  qml-module-qtquick-dialogs qml-module-qtquick-layouts
  qml-module-qtquick-privatewidgets qml-module-qtquick-window2
  qml-module-qtquick2 qt4-designer xsane-common
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 81 not upgraded.

**Comment by debanjansen48** [comment]
same issue...

**Comment by Valdes-Tresanco-MS** [comment]
What Linux distribution are you using?
Is python2 the default python in the terminal?

Run this and show me the output
```python```
```import PyQt4```

**Comment by debanjansen48** [comment]
Linux Mint 19.3 Cinnamon version 4.4.8

Linux kernal : 5.4.0-73-generic


(base) debanjan@debanjan:~$ python
Python 3.8.2 (default, Mar 26 2020, 15:53:00) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import PyQt4
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'PyQt4'



(base) debanjan@debanjan:~$ whereis python
python: /usr/bin/python3.6 /usr/bin/python3.6m /usr/bin/python /usr/bin/python3.6m-config /usr/bin/python3.6-config /usr/bin/python3.8 /usr/bin/python2.7-config /usr/bin/python2.7 /usr/lib/python3.6 /usr/lib/python3.8 /usr/lib/python3.7 /usr/lib/python2.7 /etc/python3.6 /etc/python /etc/python3.8 /etc/python2.7 /usr/local/lib/python3.6 /usr/local/lib/python3.8 /usr/local/lib/python2.7 /usr/include/python3.6 /usr/include/python3.6m /usr/include/python2.7 /usr/share/python /home/debanjan/miniconda3/bin/python3.8-config /home/debanjan/miniconda3/bin/python /home/debanjan/miniconda3/bin/python3.8 /usr/share/man/man1/python.1.gz

**Comment by Valdes-Tresanco-MS** [comment]
Ok, the error is because you have a conda environment active.
AMDock works with the system python and only with version 2. We are working on the new version with python3 since python 2 is obsolete. You can modify the AMDock.sh file in the installation folder to explicitly specify the system python2 or deactivate the conda environment

Python executable in AMDock.sh
```
# ---------------------------------------------------------------------# 
Locate a python2.7 installation directory which will be used to run the program.
# ---------------------------------------------------------------------
PYTHON="python"
```

**Comment by debanjansen48** [comment]
Hureee
thank you....Its working....

Thanks a lot.

One silly thing I would like to ask you if u don't mistake me...shall I convert protein.pdb file into protein.pdbqt file using this program?

**Comment by Valdes-Tresanco-MS** [comment]
Not necessarily, AMDock supports both the PDB file to generate the PDBQT and a PDBQT file. Typically, users enter a PDB since AMDock carries out the protonation and preparation. However, if necessary you can use the PDBQT prepared in ADT or Openbabel

**Comment by debanjansen48** [comment]
why I am receiving this error "Prepare_gpf4zn finalized with exitcode 1 and existstatus 0"

can dock set of ligands or a batch or ligands with this tool??

**Comment by Valdes-Tresanco-MS** [comment]
It could be a bug, I will check it as soon as possible.

Unfortunately no, currently AMDock does not support virtual screening. However, you can prepare the receptor with AMDock and then each ligand with a script. The receptor's pdbqt file can be found in the input folder inside of project folder

**Comment by debanjansen48** [comment]
ok, thank you.

**Comment by debanjansen48** [comment]
another error
PDB2PQR finalized with exitcode 2 and existstatus 0

the log given bellow
elcome to AMDock
Version 1.5.2 For Windows and Linux

AMDOCK: Defining Initial Parameters...
AMDOCK: DOCKING_PROGRAM: AutoDock Vina
AMDOCK: WDIR: /home/debanjan/Documents/ssp/AMDock/tst-vina
AMDOCK: MODE: SCORING
AMDOCK: TARGET: 4qu3-prep
AMDOCK: TARGET (Hetero): []
AMDOCK: TARGET (Zn atoms): []
AMDOCK: LIGAND: 4qu3-lig
AMDOCK: LIGAND (heavy_atoms): 33
AMDOCK: Defining Initial Parameters... Done 
AMDOCK: Prepare Initial Files...
AMDOCK: Running PDB2PQR for Target...

Usage: pdb2pqr.py [options] PDB_PATH PQR_OUTPUT_PATH

pdb2pqr.py: error: Unable to find file /home/debanjan/Documents/ssp/AMDock/tst-vina/input/4qu3-prep.pdb!

--------------------------
PDB2PQR - a Python-based structural conversion utility
--------------------------
Please cite your use of PDB2PQR as:
  Dolinsky TJ, Nielsen JE, McCammon JA, Baker NA.
  PDB2PQR: an automated pipeline for the setup, execution,
  and analysis of Poisson-Boltzmann electrostatics calculations.
  Nucleic Acids Research 32 W665-W667 (2004).


AMDOCK: MODE: SCORING
AMDOCK: TARGET: 4qu3-prep
AMDOCK: TARGET (Hetero): []
AMDOCK: TARGET (Zn atoms): []
AMDOCK: LIGAND: 4qu3-lig
AMDOCK: LIGAND (heavy_atoms): 33
AMDOCK: Defining Initial Parameters... Done 
AMDOCK: Prepare Initial Files...
AMDOCK: Running PDB2PQR for Target...

Usage: pdb2pqr.py [options] PDB_PATH PQR_OUTPUT_PATH

pdb2pqr.py: error: Unable to find file /home/debanjan/Documents/ssp/AMDock/tst-vina/input/4qu3-prep.pdb!

--------------------------
PDB2PQR - a Python-based structural conversion utility
--------------------------
Please cite your use of PDB2PQR as:
  Dolinsky TJ, Nielsen JE, McCammon JA, Baker NA.
  PDB2PQR: an automated pipeline for the setup, execution,
  and analysis of Poisson-Boltzmann electrostatics calculations.
  Nucleic Acids Research 32 W665-W667 (2004).


AMDOCK: MODE: SCORING
AMDOCK: TARGET: 3ni9_1prohc
AMDOCK: TARGET (Hetero): []
AMDOCK: TARGET (Zn atoms): []
AMDOCK: LIGAND: 4qu3-lig
AMDOCK: LIGAND (heavy_atoms): 33
AMDOCK: Defining Initial Parameters... Done 
AMDOCK: Prepare Initial Files...
AMDOCK: Running Prepare_Ligand4 for Ligand...set verbose to  True
read  /home/debanjan/Documents/ssp/AMDock/tst-vina/input/4qu3-lig.pdb
setting up LPO with mode= automatic and outputfilename=  None
and check_for_fragments= False
and bonds_to_inactivate= 
returning  0
No change in atomic coordinates

AMDOCK: Prepare Ligand...Done.
AMDOCK: Prepare Initial Files...Done. 
AMDOCK: MODE: SIMPLE
AMDOCK: TARGET: 4qu3-prep
AMDOCK: TARGET (Hetero): []
AMDOCK: TARGET (Zn atoms): []
AMDOCK: LIGAND: 4qu3-lig
AMDOCK: LIGAND (heavy_atoms): 33
AMDOCK: Defining Initial Parameters... Done 
AMDOCK: Prepare Initial Files...
AMDOCK: Running PDB2PQR for Target...

Usage: pdb2pqr.py [options] PDB_PATH PQR_OUTPUT_PATH

pdb2pqr.py: error: Unable to find file /home/debanjan/Documents/ssp/AMDock/tst-vina/input/4qu3-prep.pdb!

--------------------------
PDB2PQR - a Python-based structural conversion utility
--------------------------
Please cite your use of PDB2PQR as:
  Dolinsky TJ, Nielsen JE, McCammon JA, Baker NA.
  PDB2PQR: an automated pipeline for the setup, execution,
  and analysis of Poisson-Boltzmann electrostatics calculations.
  Nucleic Acids Research 32 W665-W667 (2004).


AMDOCK: MODE: SIMPLE
AMDOCK: TARGET: 4qu3-prep
AMDOCK: TARGET (Hetero): []
AMDOCK: TARGET (Zn atoms): []
AMDOCK: LIGAND: 4qu3-lig
AMDOCK: LIGAND (heavy_atoms): 33
AMDOCK: Defining Initial Parameters... Done 
AMDOCK: Prepare Initial Files...
AMDOCK: Running PDB2PQR for Target...

Usage: pdb2pqr.py [options] PDB_PATH PQR_OUTPUT_PATH

pdb2pqr.py: error: Unable to find file /home/debanjan/Documents/ssp/AMDock/tst-vina/input/4qu3-prep.pdb!

--------------------------
PDB2PQR - a Python-based structural conversion utility
--------------------------
Please cite your use of PDB2PQR as:
  Dolinsky TJ, Nielsen JE, McCammon JA, Baker NA.
  PDB2PQR: an automated pipeline for the setup, execution,
  and analysis of Poisson-Boltzmann electrostatics calculations.
  Nucleic Acids Research 32 W665-W667 (2004).

**Comment by Shredderroy** [comment]
I have similar problems. First of all, I cannot run ./install.sh. I get the following error:

`
Traceback (most recent call last):
  File "/home/aroy/Applications/AMDock-master/module_checker.py", line 24, in <module>
    obmayor, obminor, obrel = obabel_exe.communicate()[0].split()[2].split('.')
TypeError: a bytes-like object is required, not 'str'
`

Moreover, I pip is constantly complaining that Python2.7 is no longer supported. Are there any plans to upgrade the version of Python?

I am using Fedora 34, with all the latest packages installed.

**Comment by Valdes-Tresanco-MS** [comment]
Hi @debanjansen48. Sorry for the delay. 

My PhD commitments keep me running a marathon and really, I completely forgot about the report.

I can't reproduce the error. I thought it was the file name that has the "-" character. However, it seems to work well for me. Can you try removing or changing these characters to "_" ​from the file names?

Can you tell me if the files are in that directory? It could happen that the directory is not changed or the file cannot be copied to the input folder, which should show an error.
You can also run AMDock in a console, which will display the debugging data. Just run the AMDock.sh file found inside the installation directory. You send me this information, I promise to take care of the error as soon as possible

**Comment by Valdes-Tresanco-MS** [comment]
Hi @Shredderroy. Thanks for report this error.
Can you tell me what version of python you are using?
This error seems to be related to the python that is by default in the PATH.
Make sure the python executable that `install.sh` is using is the python2 executable. It currently uses the system python, so it is probably Python3 since modern OS already uses it by default.

Unfortunately, this version only works with Python2. We are working on the version with Python3. However, since the design of the program was not good (difficult to maintain, update or implement new functions) we decided to improve it. This will delay a bit longer than expected, but I think it will be worth it. Another and more important thing, AMDock2 is not funded by any project, but what time we can dedicate to it are the hours of rest. Already the doctorate is a considerable work, I also have two students to advise and my thesis project. We are making a huge effort to launch AMDock2, so we ask our users for a little patience. We are very excited to launch it as according to our ambition and prospects, it will be great.

Best!
Mario S.

**Comment by Shredderroy** [comment]
Hi @Valdes-Tresanco-MS,

Thank you very much for such a prompt response. I fully understand your commitments, and I wish you the best with your studies.

I created a new virtualenv called amdock-env in Fedora 34, in which I installed Python 2.7. So, when I activate this virtual environment and type `python` at the prompt, I get Python 2.7.

I would like to add that the Windows 10 version seems to run without any error.

Thanks again for your work in making this tool available to us at no cost!

Best wishes,

A. R.
